paddle/phi/core/distributed/auto_parallel/dist_attr.h

```diff
@@ -109,6 +158,10 @@ class TensorDistAttr {
   int64_t batch_dim_{0};
   std::vector<bool> dynamic_dims_;
   std::map<std::string, bool> annotated_;
+  // partial map would be small (less than mesh.size)
+  // iterate operation (copy and comparision) would more frequency than random
+  // element access. <key: dim on mesh, value: partial object>
+  paddle::flat_hash_map<int64_t, _Partial_> partial_status_;
```

Comment:
Why use a map structure here? If the "dim_" in Partial indicates the mesh dim, it seems unnecessary to store another mesh dim. In addition, if one tensor has only one reduce type, is it better to use a data structure like:
Partial {
vector<int64_t> mesh_dims;
ReduceType type_;
}

Reply:
Correct ！ 
firstly I thought we would use 『set』 for partial_status_, so build the Partial struct. 
then I found『map』would be better for partial_status_, in most of use cases we use dim as key to retrieve Partial .